app/utilities/PersistenceManager.java

```diff
@@ -54,6 +54,28 @@ public User getUserFromSession(UserSession userSession){
         return null;
     }
 
+    public static void editRestaurant(int restaurantId, Restaurant tempRestaurant) {
+        Restaurant restaurant = Restaurant.find.where().eq("restaurantId", restaurantId).findUnique();
+        restaurant.setDeals(tempRestaurant.getDeals());
+        restaurant.setImage(tempRestaurant.getImage());
+        restaurant.setName(tempRestaurant.getName());
+        restaurant.setPhone(tempRestaurant.getPhone());
+        restaurant.setRating(tempRestaurant.getRating());
+        restaurant.setReservationPrice(tempRestaurant.getReservationPrice());
+        restaurant.setWorkingHours(tempRestaurant.getWorkingHours());
+
+        restaurant.getAddress().setStreetName(tempRestaurant.getAddress().getStreetName());
+        restaurant.getAddress().setCity(tempRestaurant.getAddress().getCity());
+        restaurant.getAddress().setCountry(tempRestaurant.getAddress().getCountry());
```

Comment:
Maybe you should consider creating methods in Restaurant model for updating different parts: updateAddress(Address newAddress), updateDetails(Restaurant newRestaurant), updateCoordinates(Coordinates newCoordinates) ...